llama_index/memory/types.py

```diff
@@ -0,0 +1,44 @@
+from abc import abstractmethod
+from pydantic import BaseModel
+from typing import List, Optional
+
+from llama_index.llms.base import ChatMessage, LLM
+
+
+class BaseMemory(BaseModel):
+    """Base class for all memory types."""
+
+    @classmethod
+    @abstractmethod
+    def from_defaults(
+        cls,
+        chat_history: Optional[List[ChatMessage]] = None,
+        llm: Optional[LLM] = None,
+    ) -> "BaseMemory":
+        """Create a chat memory from defualts."""
+        pass
+
+    @abstractmethod
+    def get(self) -> List[ChatMessage]:
+        """Get chat history."""
```

Comment:
i imagine this interface might change in the future, as in we might want to add args to this call to control which memories are fetched in the `get`. Right now since there's no arguments it seems primarily based on recency

Reply:
true true. Just tried to keep it basic for now :)